### DISCUSSION TITLE
Why does <h2> have anchors but not <h1>?

### DISCUSSION THREAD

**Discussion by olynch** [quarto-dev/quarto-cli#7687] [Q&A]
### Description

I want to have anchors for both h1 and h2. Or alternatively, just for h2, but then I want to be able to number theorems/propositions based on section rather than chapter.

**Comment by mcanouil** [comment]
Note that H1 is used for `title`, so it is a bit "weird" to have multiple H1 headings.
I am not sure we want the title to have an anchor, thus H1 headings not having anchors.

Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````


---

Edit: here is a reproducible example

```qmd
---
title: "Quarto Playground"
format: html
anchor-sections: true
toc: true
---

This is a playground for Quarto.

# H1 Heading

## H2 Heading
```

**Reply by cderv** [reply]
Historically, `h1` have no anchor since day-1 of the feature : 4026f2dd827c5bb3316c3d6b602c820ee64944cd

> Note that H1 is used for title, so it is a but "weird" to have multiple H1 headings.
> I am not sure we want the title to have an anchor, thus H1 headings not having anchors.

I believe this is the reason why. 

We could probably adapt our selectors to get all h1 except the main title one but [this is not considered best practice to have several h1 in a document](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#avoid_using_multiple_h1_elements_on_one_page) 

So the feature is aligned with this best practice I would say of one h1 (for the title of doc) with no anchor.

**Reply by olynch** [reply]
I'm fine with just using `h2`, but in that case I'd like to prefix my numbering of theorems by the section number, rather than the chapter number. Is that possible?

**Reply by mcanouil** [reply]
It depends of your output format, in $\LaTeX$ there are various ways to do this, such as using the `\renewcommand` to change the default numbering (documentation can easily be found online).

**Reply by olynch** [reply]
I'm interested in doing this for html.

**Reply by mcanouil** [reply]
I am not sure it is possible in HTML as most of this is internal to Quarto.
There are tricks and hacks, but not great on the long run.
